sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -226,18 +226,33 @@ case class GenerateExec(
       "0"
     }
     val numOutput = metricTerm(ctx, "numOutputRows")
+    val requiredInput =
+      evaluateRequiredVariablesExpr(child.output,
+        input, parent.inputSet) ++ position ++ values
     s"""
        |${data.code}
        |$initMapData
        |int $numElements = ${data.isNull} ? 0 : ${data.value}.numElements();
        |for (int $index = $init; $index < $numElements; $index++) {
        |  $numOutput.add(1);
        |  $updateRowData
-       |  ${consume(ctx, input ++ position ++ values)}
+       |  ${consume(ctx, requiredInput)}
        |}
      """.stripMargin
   }
 
+  /**
+   * Returns [ExprCode] for required attributes
+   */
+  private def evaluateRequiredVariablesExpr(
```

Comment:
@cloud-fan Should this method be in introduced in WholeStageCodeGenExec.scala?

Reply:
How about inlining this method in the caller side? Btw, why do we need this method? It seems https://github.com/apache/spark/commit/b5c5bd98ea5e8dbfebcf86c5459bdf765f5ceb53 doesn't have this logic though.

Reply:
This method is required when we either project all column or project none of the columns(except for the exploded columns).
There were couple of test failure
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/123751/testReport/

Reply:
made the code inline.